src/main/java/rx/Observable.java

```diff
@@ -1771,8 +1775,8 @@ public void call(Subscriber<? super Object> subscriber) {
     @Experimental
     @SuppressWarnings({"unchecked", "rawtypes"})
     public final static <T> Observable<T> merge(Observable<? extends Observable<? extends T>> source, int maxConcurrent) {
-        if (source.getClass() == ScalarSynchronousObservable.class) {
-            return ((ScalarSynchronousObservable<T>)source).scalarFlatMap((Func1)UtilityFunctions.identity());
+        if (source instanceof ScalarSynchronousObservable) {
+            return ((ScalarSynchronousObservable<Observable<T>>)source).get();
```

Comment:
not sure if this is okay for the `ScalarSynchronousLazyObservable` since it makes it not so lazy.